docs/3-write-ten-dapp/4-smart-contract-features/1-data-access.md

```diff
@@ -0,0 +1,251 @@
+---
+sidebar_position: 1
+---
+
+# Data Access
+
+Solidity encapsulates "private" state variables in a way that prevents them from being accessed by other contracts.
+
+However, private variables are readable externally via [getStorageAt](https://docs.alchemy.com/reference/eth-getstorageat), meaning contracts can't truly have secrets.
+
+TEN overcomes this limitation by offering "Data Access Control" primitives.
+
+To achieve this, we had to:
+
+1. Run the Ethereum Virtual Machine inside a Trusted Execution Environment (TEE) and store the state in encrypted storage. This prevents the node operator from accessing the data.
+2. Disable the `getStorageAt` method, so now contracts can have secrets (as long as the `private` state variables are not exposed via public view functions). This prevents anyone with RPC access to the node from reading the data.
+3. Authenticate "view function calls" so that the smart contract knows who is calling the view function. Without this feature there is no "Data Access **Control**", because the dApp developer can't write access control logic. 
+4. Event logs are another way to expose data from a contract to the outside world. A practical platform needs a way to configure who can read the various event logs.
+5. Access to "Transaction Receipts", which contain the logs and the status of the transaction, must also be controlled. 
```

Comment:
The tense of this point is slightly different to the previous four. It should probably be something like 

`Control access to the "Transaction Receipts", which contain the logs and status of the transaction.`